microsoft-edge/devtools-guide-chromium/memory-problems/memory-101.md

```diff
@@ -23,165 +23,146 @@ ms.date: 07/21/2023
 
-![Node #10 is dominated by node #7 and doesn't dominate any nodes](./memory-101-images/dominators-04.png)
- 
-Node `#11` is dominated by node `#1` and doesn't dominate any nodes:
+<!-- ====================================================================== -->
+## Retainers
 
-![Node #11 is dominated by node #1 and doesn't dominate any nodes](./memory-101-images/dominators-05.png)
+An object's retainers are the others objects that hold references to it. The **Retainers** section of the **Memory** tool shows the objects that hold references to the selected object. Any object with no retainers can be discarded by the browser’s garbage collector, which reduces memory usage.
 
 
 <!-- ====================================================================== -->
 ## V8 specifics
 
-When profiling memory, it is helpful to understand why heap snapshots look a certain way.  This section describes some memory-related topics specifically corresponding to the _V8 JavaScript virtual machine_ (abbreviated here as _V8 VM_, or just _VM_).
+When profiling memory, it is helpful to understand why heap snapshots look a certain way. This section describes some memory-related topics specifically corresponding to the _V8 JavaScript virtual machine_ (abbreviated here as _V8 VM_, or just _VM_).
 
-### JavaScript object representation
+<!-- ------------------------------ -->
+#### JavaScript object representation
 
 In JavaScript, there are three primitive types:
 
 *  Numbers (such as `3.14159...`).
 *  Booleans (`true` or `false`).
 *  Strings (such as `"Werner Heisenberg"`).
 
-Primitives cannot reference other values, and are always leaf nodes (also called _terminating nodes_).
+Primitives cannot reference other values, and are always leaf nodes in the memory graph (also called _terminating nodes_).
 
 **Numbers** can be stored as either:
 
 *  Immediate 31-bit integer values that are called **small integers** (_SMIs_).
 
-*  Heap objects, referred to as **heap numbers**.  Heap numbers are used for storing values that don't fit into the SMI form, such as **doubles**, or when a value needs to be **boxed**, such as setting properties on it.
+*  Heap objects, referred to as **heap numbers**. Heap numbers are used for storing values that don't fit into the SMI form, such as **doubles**, or when a value needs to be **boxed**, such as setting properties on it.
 
 **Strings** can be stored in either:
 
 *  The **VM heap**.
 
-*  Externally in the **renderer's memory**.  A _wrapper object_ is created and used for accessing external storage where, for example, script sources and other content that is received from the Web is stored, rather than copied onto the VM heap.
+*  Externally in the **renderer's memory**. A _wrapper object_ is created and used for accessing external storage where, for example, script sources and other content that is received from the Web is stored, rather than copied onto the VM heap.
```

Comment:
```suggestion
*  Externally in the **renderer's memory**. A _wrapper object_ is created and used for accessing external storage where, for example, script sources and other content that's received from the Web is stored, rather than copied onto the VM heap.
```